test/testsuites/packagestests.go

```diff
@@ -102,7 +112,116 @@ func testOpSrcWithConflictingPackages(t *testing.T) {
 	err = helpers.CreateRuntimeObject(test.Global.Client, ctx, helpers.CreateOperatorSourceDefinition(opSrcName, namespace))
 	assert.NoError(t, err, "Could not create operator source")
 
+	// Check that the Deployment was created.
+	resultDeployment := &apps.Deployment{}
+	err = helpers.WaitForResult(client, resultDeployment, namespace, opSrcName)
+	assert.NoError(t, err)
+
+	// Now check that the Deployment is ready.
+	err = helpers.WaitForSuccessfulDeployment(client, *resultDeployment)
+	assert.NoError(t, err)
+
+	// Check that a CSC with an unspecified source cannot be resolved if its
+	// packages exist in two OperatorSources.
+	packages := "camel-k-marketplace-e2e-tests"
+	runSourceTest(t,
+		"",
+		packages,
+		"Configuring",
+		fmt.Sprintf("Unable to resolve the source - multiple sources contain the requested packages [%s]", packages),
+	)
+
 	// Confirm child resources were created without errors.
 	err = helpers.CheckCscChildResourcesCreated(client, opSrcName, namespace, namespace)
 	assert.NoError(t, err, "Child resources not created")
 }
+
+// resolveMissingSource ensures that if no source is given, the CatalogSourceConfig is
+// placed in the Succeeded phase if a source exists that contains the provided packages.
+func resolveMissingSource(t *testing.T) {
```

Comment:
I am not a huge fan of this pattern where the main test case has no asserts but it is the helper that has the asserts.

Reply:
I will update the method to return an error and fail in the main test across all instances.

Reply:
The helper has been updated to return an error.